src/server/session.ts

```diff
@@ -1537,6 +1537,11 @@ namespace ts.server {
             const position = this.getPositionInFile(args, file);
             const projects = this.getProjects(args);
 
+            const renameInfo = this.mapRenameInfo(
+                this.getRenameInfo(args), Debug.checkDefined(this.projectService.getScriptInfo(file)));
+
+            if (!renameInfo.canRename) return { info: renameInfo, locs: [] };
+

```

Comment:
1. this was added because i think we don't want to call: 
`combineProjectOutputForRenameLocations` -> `combineProjectOutputWorker` -> `findRenameLocations`
as it seems this is part of what causes the response time to scale, due to loading projects . also, before this PR `findRenameLocations` computed locations for `!nodeIsEligibleForRename` and now it doesn't.
so we could remove this condition and let it go through the original flow, but it seems unnecessary as we can already tell `canRename` is `false`.
2. should this line account for `simplifiedResult` too? (return `[]`)

Reply:
1. Good observation—project loading is super expensive, so that definitely explains why the numbers we were seeing were so bad. We definitely don’t want to load projects if we don’t have to.
2. Yes.